pkg/controller/common/rollout/workloads/cloneset_controller.go

```diff
@@ -58,8 +58,10 @@ func (c *CloneSetController) Size(ctx context.Context) (int32, error) {
 }
 
 // Verify verifies that the target rollout resource is consistent with the rollout spec
-func (c *CloneSetController) Verify(ctx context.Context) *v1alpha1.RolloutStatus {
+func (c *CloneSetController) Verify(ctx context.Context) (status *v1alpha1.RolloutStatus) {
```

Comment:
for my experiense, use a named return argument is anti-partten.

1. reviewer can hardly understand what value is really returned, they have to check code all the time.
2. it's easier to write bugs because you will ignore the return without variable.

Reply:
https://golang.org/doc/effective_go.html#named-results

I actually use the named results to make the code clean and easier to check. I feel that there are too many places that return value. Instead of checking the return value at every place, we can check the returned value at only one place.   

I wonder if that makes sense?

Reply:
This works when error return is always the same. When the named return values differ on different errors, it can be very hard to maintain.

Anyway, I'm fine with this, but the named-results always need to be reviewed cautiously. I'm afrait this is not a good start.

Reply:
The error only depends on the returned status, thus I put that logic at the only exit point instead of every return point.